products/models.py

```diff
@@ -55,8 +68,33 @@ class Product(AuditTimeStamp):
     description = models.CharField(
         _("Product Description"), max_length=1000, blank=True
     )
+
+    image = models.ImageField(upload_to=image_path, default="default_image.png")
     name = models.CharField(_("Product Name"), max_length=100)
 
+    def get_price(self):
+        """
+        Returns price with currency
+        Returns:
+            (str): Value containing price and currency
+        """
+        return f"Rs. {self.price}"
+
+    def save(self, *args, **kwargs):
+        # pylint: disable=no-member
+        if self.id is not None:
+            product = Product.objects.get(id=self.id)
+            if product.image.name != "default_image.png":
+                product.image.delete(False)
+
+        super().save(*args, **kwargs)
+
+    def delete(self, *args, **kwargs):
+        # pylint: disable=no-member
+        if self.image.name != "default_image.png":
+            self.image.delete(False)
```

Comment:
What is meant by False here?

Reply:
By default  it is `True` so if i have `self.image.delete()` it will call the `model.save()` which we dont want in this case because it will cause and infinite loop as `.save()` will also call `self.image.delete()`